src/AWS.jl

```diff
@@ -33,19 +33,39 @@ aws_config = Ref{AWSConfig}()
     global_aws_config()
 
 Retrieve the global AWS configuration.
+If one is not set, create one with default configuration options.
+
+# Keywords
+- `kwargs...`: AWSConfig kwargs to be passed along if the global configuration is not already set
 
 # Returns
 - `AWSConfig`: The global AWS configuration
 """
-function global_aws_config()
+function global_aws_config(; kwargs...)
     if !isassigned(aws_config)
```

Comment:
Should probably always create a new `AWSConfig` if `kwargs` is passed in:
```suggestion
    if !isassigned(aws_config) || !isempty(kwargs)
```